rbmc-cfam-daemon/src/cfam_access.cpp

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "cfam_access.hpp"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <phosphor-logging/lg2.hpp>
+
+#include <unordered_map>
+
+namespace fs = std::filesystem;
+
+const std::unordered_map<cfam::ScratchPadReg, std::string> scratchPadRegNames{
+    {cfam::ScratchPadReg::one, "scratch1"},
+    {cfam::ScratchPadReg::two, "scratch2"},
+    {cfam::ScratchPadReg::three, "scratch3"},
+    {cfam::ScratchPadReg::four, "scratch4"}};
+
+fs::path CFAMAccess::getRegisterPath(cfam::ScratchPadReg reg) const
+{
+    return std::format("/sys/class/fsi-master/fsi{}/slave@00:00/{}", link,
+                       scratchPadRegNames.at(reg));
+}
+
+bool CFAMAccess::exists() const
+{
+    auto path = getRegisterPath(cfam::ScratchPadReg::one);
+
+    try
+    {
+        return fs::exists(path);
+    }
+    catch (const std::exception& e)
+    {
+        lg2::error("Failed calling filesystem::exists on {FILE}: {ERR}", "FILE",
+                   path, "ERR", e);
+    }
+    return false;
+}
+
+std::expected<uint32_t, int> CFAMAccess::readScratchReg(cfam::ScratchPadReg reg)
+{
+    auto regFile = getRegisterPath(reg);
```

Comment:
We could combine lines 43 and 45, but feel free to ignore this if you plan to add additional logic between them in future commits.

Reply:
will fix.